### DISCUSSION TITLE
Backend Performance Benchmarking

### DISCUSSION THREAD

**Discussion by simonelnahas** [realworld-apps/realworld#1181] [Ideas]
Hey Community,

I've been thinking about benchmarking the various backends' performance.
What tests do you think would be interesting to see?

Do you have any stress/load testing frameworks/tools that you can recommend?

Best,
Simon

**Comment by simonelnahas** [comment]
My plan is to perform the benchmarking on a few backends and then open-source the tests and how to do them, such that all of the backends can test the performance of their implementation of the RealWorld API.

**Comment by geromegrignon** [comment]
Hi Simon, about recommended tools, I won't be able to help that much as I'm unfamiliar with such testing.

About stress/load testing, please note it requires a production environment with enough resources for the test to be efficient. Depending on the server/database providers available for the implementation, it might mean extra fees. That's why most backend implementations are not deployed, while there are free easy solutions for static frontend implementations.

**Comment by geromegrignon** [comment]
For load testing, you can take a look at this pull request though: https://github.com/gothinkster/realworld/pull/562
